runtime/server/controller.go

```diff
@@ -268,23 +268,57 @@ func (s *Server) CreateTrigger(ctx context.Context, req *runtimev1.CreateTrigger
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
-	var kind string
-	r := &runtimev1.Resource{}
+	// Backwards compatibility for req.Trigger (deprecated)
+	if req.Trigger != nil {
+		switch trg := req.Trigger.(type) {
+		case *runtimev1.CreateTriggerRequest_PullTriggerSpec:
+			req.Parser = true
+		case *runtimev1.CreateTriggerRequest_RefreshTriggerSpec:
+			req.Resources = trg.RefreshTriggerSpec.Resources
+			req.Models = trg.RefreshTriggerSpec.Models
+		}
+	}
+
+	// Build refresh trigger spec
+	spec := &runtimev1.RefreshTriggerSpec{
+		Resources: req.Resources,
+		Models:    req.Models,
+	}
 
-	switch trg := req.Trigger.(type) {
-	case *runtimev1.CreateTriggerRequest_PullTriggerSpec:
-		kind = runtime.ResourceKindPullTrigger
-		r.Resource = &runtimev1.Resource_PullTrigger{PullTrigger: &runtimev1.PullTrigger{Spec: trg.PullTriggerSpec}}
-	case *runtimev1.CreateTriggerRequest_RefreshTriggerSpec:
-		kind = runtime.ResourceKindRefreshTrigger
-		r.Resource = &runtimev1.Resource_RefreshTrigger{RefreshTrigger: &runtimev1.RefreshTrigger{Spec: trg.RefreshTriggerSpec}}
+	// Handle the convenience flag for the project parser.
+	if req.Parser {
+		spec.Resources = append(spec.Resources, runtime.GlobalProjectParserName)
 	}
 
-	n := &runtimev1.ResourceName{
-		Kind: kind,
-		Name: fmt.Sprintf("trigger_adhoc_%s", time.Now().Format("200601021504059999")),
+	// Handle the convenience flags for all sources and models.
+	if req.AllSourcesModels || req.AllSourcesModelsFull {
+		// Add all sources.
+		// Note: Don't need to handle "full" here since source refreshes are always full refreshes.
+		rs, err := ctrl.List(ctx, runtime.ResourceKindSource, "", false)
+		if err != nil {
+			return nil, status.Error(codes.InvalidArgument, fmt.Errorf("failed to list sources: %w", err).Error())
+		}
+		for _, r := range rs {
+			spec.Resources = append(spec.Resources, r.Meta.Name)
+		}
+
+		// Add all models.
+		rs, err = ctrl.List(ctx, runtime.ResourceKindModel, "", false)
+		if err != nil {
+			return nil, status.Error(codes.InvalidArgument, fmt.Errorf("failed to list models: %w", err).Error())
+		}
+		for _, r := range rs {
+			spec.Models = append(spec.Models, &runtimev1.RefreshModelTrigger{
+				Model: r.Meta.Name.Name,
+				Full:  req.AllSourcesModelsFull,
+			})
+		}
 	}
 
+	// Create the trigger resource
+	name := fmt.Sprintf("trigger_adhoc_%s", time.Now().Format("200601021504059999"))
```

Comment:
What is this format ? Can we add a constant so that we get a hint from the name or add a comment here?

Reply:
Done